src/User.php

```diff
@@ -2923,7 +2925,10 @@ public function showForm($ID, array $options = [])
                 $entities = $this->getEntities();
                 Entity::dropdown(['value'  => $this->fields["entities_id"],
                     'rand'   => $entrand,
-                    'entity' => $entities
+                    'entity' => $entities,
+                    'toadd' => [
+                        0 => __('Full structure'),
```

Comment:
The `0` id is already used by the root entity.

This mean with this change I can't pick the root entity as my default anymore.
Is it on purpose ?

Reply:
It always has been this way

Reply:
If you have access to the root entity, the behavior is the same between "full structure" and "tree structure." The difference only occurs when you do not have access to the root entity.

We cannot use `-1` without modifying the database schema, and I'm not sure it's worth it?

Reply:
What do you mean ?
Before this change, I can pick the root entity. 
After this change, I can't.

Reply:
> If you have access to the root entity, the behavior is the same between "full structure" and "tree structure." The difference only occurs when you do not have access to the root entity.

Maybe the root entity option should be removed then ?

Because right now if I click on the root entity it selects full structure instead.
It is likely that we will get support requests from people that will think it is a bug :/

Reply:
The `0 => "Root entity"` value should indeed be removed to prevent having two options with the `0` value. Indeed, the `select2` component does not handle it properly.

Same remark for the `showMyForm()` method.